src/devices/tfa_drop_30.3233.01.c

```diff
@@ -0,0 +1,307 @@
+- 1 is encoded as 500 us pulse and a 250us gap
+
+Note that this encoding scheme is inverted relative to the default
+interpretation of short/long pulses in the PWM decoder in rtl_433.
+The implementation below thus inverts the buffer. The protocol is
+described below in the correct space, i.e. after the buffer has been
+inverted.
+
+Not every tip of the bucket triggers a message immediately. In some
+cases, artifically tipping the bucket many times lead to the base
+station ignoring the signal completely until the device was reset.
+
+Data layout:
+
+```
+CCCCIIII IIIIIIII IIIIIIII BCUU XXXX RRRRRRRR CCCCCCCC SSSSSSSS MMMMMMMM
+KKKK
+```
+
+- C: 4 bit message prefix, always 0x3
+- I: 2.5 byte ID
+- B: 1 bit, battery_low. 0 if battery OK, 1 if battery is low.
+- C: 1 bit, device reset. Set to 1 briefly after battery insert.
+- X: Transmission counter
+     - Possible values: 0x0, 0x2, 0x4, 0x6, 0x8, 0xA, 0xE, 0xE.
+     - Rolls over.
+- R: LSB of 16-bit little endian rain counter
+- S: MSB of 16-bit little endian rain counter
+- C: Fixed to 0xaa
+- M: Checksum.
+   - Compute with reverse Galois LFSR with byte reflection, generator
+     0x31 and key 0xf4.
+   - See `compute_checksum` below.
+- K: Unknown. Either b1011 or b0111.
+     - Distribution: 50:50.
+
+[Bitbench](http://triq.net/bitbench) string:
+
+```
+ID:hh ID:hh ID:hh BAT_LOW:b RESET:b UNKNOWN:bb XMIT_COUNTER:h RAIN_A:d
+CONST:hh RAIN_B:d CHECK:8b UNKNOWN:bbxx xxxx
+```
+
+Some example data:
+
+```
+c240aaff09550021c
+c240aabf095500e04
+c240aafd095500b64
+c240aafb0955003e4
+c240aaf9095500a9c
+c212b7f9035500e5c
+c212b7f703550053c
+c212b7f5035500c44
+```
+
+
+The rain bucket counter represents the number of tips of the rain
+bucket. Each tip of the bucket corresponds to 0.254mm of rain.
+
+The rain bucket counter does not start at 0. Instead, the counter
+starts at 65526 to indicate 0 tips of the bucket. The counter rolls
+over at 65535 to 0, which corresponds to 9 and 10 tips of the bucket.
+
+If no change is detected, the sensor will continue broadcasting
+identical values. This lasts at least for 20 minutes,
+potentially forever.
+
+The second nibble of byte 3 is a transmission counter: 0x0, 0x2, 0x4,
+0x6, 0x8, 0xa, 0xc, 0xe. After the transmission with counter 0xe, the
+counter rolls over to 0x0 on the next transmission and the cycle starts
+over.
+
+After battery insertion, the sensor will transmit 7 messages in rapid
+succession, one message every 3 seconds. After the first message,
+the remaining 6 messages have bit 1 of byte 3 set to 1. This could be
+some sort of reset indicator.
+For these 6 messages, the transmission counter does not increase.
+
+After the full 7 messages, one regular message is sent after 30s.
+Afterwards, messages are sent every 45s.
+*/
+
+#include "decoder.h"
+
+#define TFA_DROP_BITLEN 66
+#define TFA_DROP_STARTBYTE 0x3 /* Inverted already */
+#define TFA_DROP_MINREPEATS 2
+
+static uint8_t compute_checksum(uint8_t *msg, int bytes);
+
```

Comment:
Just a tiny nit not needed for merge. Move the function here to skip the future declaration part. Or move the checksum calculation to the util file.